diagnostics/diag.go

```diff
@@ -336,3 +346,16 @@ func (d *Diagnostics) handleNewStream(s inet.Stream) {
 	d.HandleMessage(context.Background(), s)
 	s.Close()
 }
+
+func GetCommitHash() (string, error) {
+	gopath := os.Getenv("GOPATH")
+	ipfs := gopath + "/src/github.com/jbenet/go-ipfs"
```

Comment:
This is worse. it will fail mysteriously. binaries must not report to be a version they are not.